packages/web-client/app/components/card-pay/issue-prepaid-card-workflow/funding-source/index.ts

```diff
@@ -0,0 +1,91 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { reads } from 'macro-decorators';
+import { inject as service } from '@ember/service';
+import { taskFor } from 'ember-concurrency-ts';
+import { task } from 'ember-concurrency-decorators';
+import Layer2Network from '@cardstack/web-client/services/layer2-network';
+import WorkflowSession from '../../../../models/workflow/workflow-session';
+import BN from 'bn.js';
+import { toBN } from 'web3-utils';
+import {
+  TokenDisplayInfo,
+  TokenSymbol,
+  bridgeableSymbols,
+} from '@cardstack/web-client/utils/token';
+
+interface Token {
+  balance: BN;
+  icon: string;
+  name: string;
+  description: string;
+  symbol: TokenSymbol;
+}
+
+interface FundingSourceCardArgs {
+  workflowSession: WorkflowSession;
+  onComplete: (() => void) | undefined;
+  onIncomplete: (() => void) | undefined;
+  isComplete: boolean;
+}
+
+class FundingSourceCard extends Component<FundingSourceCardArgs> {
+  tokenInfo = bridgeableSymbols.map((symbol) => new TokenDisplayInfo(symbol));
+  @service declare layer2Network: Layer2Network;
+
+  @reads('args.workflowSession.state.prepaidFundingToken')
+  declare selectedToken: Token;
+
+  constructor(owner: unknown, args: FundingSourceCardArgs) {
+    super(owner, args);
+    taskFor(this.fetchDepotTask)
+      .perform()
+      .then((depot: { address: string; tokens: [] }) => {
+        if (depot) {
+          this.args.workflowSession.update('depotAddress', depot.address);
+          this.args.workflowSession.update('depotTokens', depot.tokens);
+        }
+      })
+      .then(() => {
+        if (!this.selectedToken) this.chooseSource(this.tokens[0]);
+      });
```

Comment:
Maybe we should move the code within these `then` blocks into `fetchDepotTask`?

Reply:
These should probably be managed within the `Layer2Chain` class, since our balances there should come from the depot too.

Reply:
I only had to do it here because it was easier to fetch the real values than create fake ones that didn't error for the UI. Should I do the layer2chain strategy stuff in this PR? I was thinking of refactoring separately since it's kinda hard to ship this UI without any data. I also saw that you created a similar ticket, hope that's not the same thing

Reply:
I think a separate PR is good. I'll work on fetching depot balances for the layer two connect card, and make the necessary changes to the use of depots here and in the other cards that use it.

Reply:
if I move this line to fetchDepot, then the dropdown displays 0 balance. I moved it up to the previous block, that works

Reply:
Draft implemented in #1686. Some changes to come + perhaps restructuring, but:

We fetch the depot and store in layer 2 web3 strategy, because we need the depot in order to get the balances. The depot balances are the same as the layer 2 chain balances, so you can get them directly from `this.layer2Network`. As a result, all uses of `fetchDepotTask` should be refactored out in #1686.